File: ghost-coach-mobile/lib/data/model/session/confidence_level.dart
enum ConfidenceLevel {
  low('LOW', 'Low'),
  medium('MEDIUM', 'Medium'),
  high('HIGH', 'High');

  const ConfidenceLevel(this.wire, this.label);

  final String wire;
  final String label;

  static ConfidenceLevel fromWire(String? value) {
    if (value == null) return ConfidenceLevel.low;
    for (final v in values) {
      if (v.wire == value) return v;
    }
    return ConfidenceLevel.low;
  }
}

File: ghost-coach-be/src/main/java/com/playmotech/ghostcoach/session/ConfidenceLevel.java
package com.playmotech.ghostcoach.session;

public enum ConfidenceLevel {
    LOW, MEDIUM, HIGH;

    public static ConfidenceLevel fromStringOrLow(String value) {
        if (value == null) return LOW;
        try {
            return ConfidenceLevel.valueOf(value.trim().toUpperCase());
        } catch (IllegalArgumentException e) {
            return LOW;
        }
    }
}
